linkerd/app/outbound/src/lib.rs

```diff
@@ -119,7 +136,27 @@ impl<S> Outbound<S> {
         P::Error: Send,
         I: io::AsyncRead + io::AsyncWrite + io::PeerAddr + std::fmt::Debug + Send + Unpin + 'static,
     {
-        let http = self
+        // HTTP per-endpoint stack used when a profile is not discovered.
+        let http_endpoint = self
+            .clone()
+            .push_tcp_endpoint()
+            .push_http_endpoint()
+            .push_into_endpoint()
+            .push_http_server::<http::Accept, _>()
+            .into_inner();
+
+        // HTTP and TCP per-endpoint stack used when a profile is not discovered.
+        let endpoint = self
+            .clone()
+            .push_tcp_endpoint()
+            .push_tcp_forward()
+            .push_into_endpoint::<(), tcp::Accept>()
+            // If HTTP is detected, use the `http_endpoint` stack
+            .push_detect_http(http_endpoint)
```

Comment:
If HTTP is detected, then we'd use the `http_endpoint` stack. That already has `push_into_endpoint` and `push_tcp_endpoint` on it though (lines 144 and 142). Can those be removed or changed in a way that we don't need to have those additional calls in this `endpoint` stack?

Reply:
we could call `push_tcp_endpoint` once and then clone it, but we can't reuse the `push_into_endpoint` call, because it's _before_ the TCP forward and HTTP endpoint stacks, which differ.

Reply:
sorry, never mind --- turns out this breaks type inference